### PR TITLE
Update Import button and add further workflow clarifications

### PR DESCRIPTION
**Pull request by rosteen**
Changes the text of the Import button and adds some clarifying text in other related places in the UI. I made an effort to update the screenshots and text in the docs that are directly related to this, but there are plenty of other screenshots that are already outdated for other reasons, so I didn't comprehensively change every screenshot in the docs here. 

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set?
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
Just to be sure, that Firefox bug is still an issue with this PR?

**Review by kecnry — COMMENTED** [review]
Thanks for all the docs updates too!  Just a few thoughts on the changes, but the implementation itself seems fine.

* If the text added to the data menus takes two lines anyways... would we be better just adding a button to each menu that adds data?

* Since the data menus are now filtering, it technically would be possible to load a file, but not see it in the list from that menu - loading a 1d spectrum in cubeviz, for example, won't show in the image viewers.

* [data_selected_1.png](https://github.com/spacetelescope/jdaviz/pull/1349/files?short_path=62a5ca8#diff-62a5ca8f576ff73b1899bb51f3f73360db846bdc4e2484319c6f808a42c54957) has "Load Data" in the screenshot instead of "Import Data".

**Comment by rosteen** [comment]
> Just to be sure, that Firefox bug is still an issue with this PR?

I'm completely blanking on what that bug was, but I wasn't able to reproduce it or track down a cause, so this definitely doesn't fix it.

**Comment by rosteen** [comment]
Right, the single click navigation into directories not working. What I said above was accurate, it was transient and I wasn't able to reproduce it.

**Review by kecnry — APPROVED** [review]
LGTM

Edit: just saw the RTD build is failing, but I assume that is just a small fix that won't otherwise affect the review.

File: docs/specviz/displaying.rst [hunk 1]
```diff
@@ -28,7 +28,9 @@ using the Python API.  Both are detailed further below.
 Selecting/Showing Data Sets
 ===========================
 
-Data can be selected and de-selected under the "Data" tab. Here, you can click a checkbox next to the listed data to make the data visible (checked) or invisible (unchecked).
+Data can be selected and de-selected in each viewer's data menu, opened by clicking the 
+|icon-viewer-data-select| button in the top left of the viewer. Here, you can click a 
+checkbox next to the listed data to make the data visible (checked) or invisible (unchecked).
```
Comment (pllim) [inline docs/specviz/displaying.rst:33]: I feel like this is more accurate, as we also have the 👁️ that toggles visibility. 🤷 

```suggestion
checkbox next to the listed data to make the data loaded (checked) or unloaded (unchecked).
```

Comment (pllim) [reply docs/specviz/displaying.rst:33]: Okay, maybe "loaded" is also misleading because it is loaded in the app but just not the viewer... My head hurts.

**Comment by pllim** [comment]
Dark mode strikes again!

Screenshot 2022-05-27 152056

**Review by pllim — APPROVED** [review]
LGTM. Thanks!

Screenshot 2022-06-01 134651

**Pull request merged**